### PR TITLE
Adding BFS and shortest path functionality to directed graphs

### PR DESCRIPTION
**Pull request by RyanMarcus**
I believe I kept with the idioms used for the DFS implementation as well as I could, but I might be missing something. I'm relatively new to Golang, so feel free to critique.

### CODE REVIEW

**Comment by RyanMarcus** [comment]
I believe this resolves issue #6.

File: directed/storage.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ func (graph *Graph) HasEdge(source, target Vertex) bool {
 	return false
 }
 
-// Order will return the order of the graph, that isnumber of vertices
+// Order will return the order of the graph, that is, the number of vertices
```
Comment (RyanMarcus) [inline directed/storage.go:145]: This is just a typo correction. Hadn't even realized I committed it.

Comment (mkindahl) [inline directed/storage.go:145]: Hi Ryan,

Thanks for the patch! I will take a look at it after the holidays, or
earlier if I have some time.

Best wishes,
Mats Kindahl

Comment (mkindahl) [inline directed/storage.go:145]: Hi Ryan,

I had a look at the code and it looks excellent, so I think I should
merge it into the branch.

One minor thing is that it seems that the Go community uses the
convention that all docstrings should start with the entity being
documented. See, for example,
http://golang.org/pkg/regexp/#Regexp.ReplaceAllFunc.

It's a minor thing, but I am quite new to the Go community and I usually
try to follow the conventions that are already in place. It avoids a lot
of problems down the road.

Best wishes,
Mats Kindahl

I'll merge it as On 12/26/2013 06:29 AM, RyanMarcus wrote:

Comment (RyanMarcus) [inline directed/storage.go:145]: I'll correct the docstrings once I get back from my trip.

**Comment by RyanMarcus** [comment]
Well, I really let this fall through the cracks. Better late than never, right? I've corrected the docstrings.

**Comment by mkindahl** [comment]
On 12/03/2014 04:42 PM, Ryan Marcus wrote:

> Well, I really let this fall through the cracks. Better late than
> never, right? I've corrected the docstrings.

Hi Ryan,

This is stuff we do in our spare time, so no problems. :)

I'll take a look at it ASAP and be back.

Best wishes,
Mats Kindahl

**Pull request merged**